scripts/test-crds.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+# ------------------------------------------------------------------------------
+# test-crds.sh
+#
+# This script tests that our legacy v1beta1 versions of our CRDs upgrade cleanly
+# to their v1 versions, and that both the legacy and the v1 versions of the CRDs
+# both function with our controllers.
+#
+# Test Workflow
+#
+#   Step 1
+#
+#   1. deploy the legacy CRDs
+#   2. install the chart
+#   3. verify legacy CRDs are in place
+#   4. run helm tests (via legacy CRDs)
+#
+#   Step 2
+#
+#   1. cleanup the chart
+#   2. deploy the v1 CRDs
+#   3. verify the CRDs updates
+#   4. redeploy the chart
+#   5. run helm tests (via v1 CRDs)
+#
+# NOTE: these tests are sensitive to which Kubernetes version they are run on.
+#       they will only run on Kubernetes clusters between v1.16.x and 1.21.x,
+#       as v1.15.x is too old for the new CRDs and v1.22.x is to new for the
+#       old CRDs. At some point in the future (potentially when <v1.22.x is no
+#       longer supported by the chart) we can remove these tests.
+# ------------------------------------------------------------------------------
+
+# ------------------------------------------------------------------------------
+# Configuration
+# ------------------------------------------------------------------------------
+
+TEST_ENV_NAME="${TEST_ENV_NAME:-kong-charts-tests}"
+
+set -euo pipefail
+
+# ------------------------------------------------------------------------------
+# Kubernetes Cluster Setup
+# ------------------------------------------------------------------------------
+
+kubectl cluster-info --context kind-"${TEST_ENV_NAME}" 1>/dev/null
+KUBERNETES_VERSION="$(kubectl version -o json | jq -r '.serverVersion.gitVersion')"
+
+echo "INFO: running CRD tests on Kubernetes ${KUBERNETES_VERSION}"
+
+# ------------------------------------------------------------------------------
+# Setup Chart Cleanup - Kubernetes Ingress Controller
+# ------------------------------------------------------------------------------
+
+RELEASE_NAME="chart-tests"
+RELEASE_NAMESPACE="$(uuidgen)"
+
+function cleanup() {
+    echo "INFO: cleaning up helm release ${RELEASE_NAME}"
+    helm delete --namespace "${RELEASE_NAMESPACE}" "${RELEASE_NAME}"
+    kubectl delete namespace "${RELEASE_NAMESPACE}"
+    kubectl delete clusterrole "${RELEASE_NAME}"-kong
+    exit 1
+}
+
+trap cleanup SIGINT
```

Comment:
why not trap on exit, instead of specific signals?

Reply:
That would run cleanup under normal script exit. We could make that work, but really this trap is _only_ here to cleanup in the event that something fails mid-script.

Reply:
These cleanups were helpful for debugging/development of the scripts but ultimately there's a script entirely for cleanup so these are now removed in f3968d43f00feae14859224d9f9be514d4348dd0.